src/Umbraco.Web.UI.Client/src/common/services/umbrequesthelper.service.js

```diff
@@ -15,16 +15,19 @@ function umbRequestHelper($http, $q, notificationsService, eventsService, formHe
          *
          * @description
          * This will convert a virtual path (i.e. ~/App_Plugins/Blah/Test.html ) to an absolute path
-         * 
+         *
          * @param {string} a virtual path, if this is already an absolute path it will just be returned, if this is a relative path an exception will be thrown
          */
         convertVirtualToAbsolutePath: function(virtualPath) {
-            if (virtualPath.startsWith("/")) {
-                return virtualPath;
-            }
-            if (!virtualPath.startsWith("~/")) {
-                throw "The path " + virtualPath + " is not a virtual path";
+            // This will convert a local path to virtual path (i.e. /App_Plugins/Blah/Test.html ) to an virtual path (i.e. ~/App_Plugins/Blah/Test.html)
```

Comment:
I don't really get why we're doing this?
We take the `virtualPath` and we do these to things to ensure it conforms to a format of `~/App_Plugins/` etc. 
Then we take this and trim away that part we just made sure was there (`~/`) and then concat it with the `applicationPath`.

Wouldn't it be enough to simply do this:
``` 
if (!Umbraco.Sys.ServerVariables.application.applicationPath) {
    throw "No applicationPath defined in Umbraco.ServerVariables.application.applicationPath";
}
return Umbraco.Sys.ServerVariables.application.applicationPath + virtualPath.trimStart("~").trimStart("/");
```

Reply:
Scratch that... there's something going on in this whole PR that isn't really right.